### DISCUSSION TITLE
Upgrading from v4.38 beta 2 to v4.38.6 failed

### DISCUSSION THREAD

**Discussion by Dima-Kal** [authelia/authelia#6975] [Troubleshooting] [answered]
### Version

v4.38.6

### Deployment Method

Docker

### Reverse Proxy

Traefik

### Description

Getting the following errors in the log:

`
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: session: option 'cookies' must be configured with the per cookie option 'default_redirection_url' but the global one is configured which is not supported" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: identity_providers: oidc: clients: client 'miniflux': option 'sector_identifier_uri' with value '': must be a URL with only the host component but appears to be invalid" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: identity_providers: oidc: clients: client 'outline-app': option 'sector_identifier_uri' with value '': must be a URL with only the host component but appears to be invalid" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=fatal msg="Can't continue due to the errors loading the configuration" stack="github.com/authelia/authelia/v4/internal/commands/context.go:320 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
`

Reading in the docs about "sector_identifier_uri" it says its not a required configuration, but it does not appear in my config at all so i dont know whats going on

### Reproduction

z

### Build Information

```shell
z
```

### Logs (Authelia)

```shell
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: session: option 'cookies' must be configured with the per cookie option 'default_redirection_url' but the global one is configured which is not supported" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: identity_providers: oidc: clients: client 'miniflux': option 'sector_identifier_uri' with value '': must be a URL with only the host component but appears to be invalid" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=error msg="Configuration: identity_providers: oidc: clients: client 'outline-app': option 'sector_identifier_uri' with value '': must be a URL with only the host component but appears to be invalid" stack="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
time="2024-03-20T12:59:13+02:00" level=fatal msg="Can't continue due to the errors loading the configuration" stack="github.com/authelia/authelia/v4/internal/commands/context.go:320 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:185 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1695                                         goexit"
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The errors you're encountering are related to the configuration of Authelia, specifically with the session cookies and OIDC (OpenID Connect) client configurations. Let's address them one by one:

1. **Session Cookies Configuration Error**:
   The error message indicates that the `default_redirection_url` is configured globally, which is not supported. Authelia's session configuration now requires that the `default_redirection_url` be configured per cookie rather than globally. This change aims to provide more granular control over the session behavior. You'll need to adjust your configuration to define `default_redirection_url` within the scope of each cookie configuration.

2. **OIDC Client Configuration Error**:
   The errors related to `sector_identifier_uri` for the OIDC clients 'miniflux' and 'outline-app' indicate that this option must be a URL with only the host component, but it appears to be invalid or not provided. The `sector_identifier_uri` is an optional field that is used in specific OIDC scenarios, particularly when you have multiple redirect URIs. If your OIDC client does not require this, ensure it is either correctly set or omitted if not needed.

To resolve these issues, you will need to review and update your Authelia configuration file. Ensure that for each session cookie configuration, the `default_redirection_url` is specified within the cookie's configuration block. For the OIDC clients, verify that the `sector_identifier_uri` is correctly formatted as a URL or omitted if unnecessary.

Unfortunately, without seeing your actual configuration, I can't provide a precise solution. However, these guidelines should help you adjust your configuration to resolve the errors. If you need further assistance, please provide more details about your configuration, especially the session and OIDC client sections.

Remember to review the Authelia documentation for [Session Configuration](https://www.authelia.com/configuration/session/introduction) and [OIDC Integration](https://www.authelia.com/integration/openid-connect/gitea) for more detailed information on configuring these aspects.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The errors you're encountering are related to the configuration of Authelia, specifically with the session cookies and OIDC (OpenID Connect) client configurations. Let's address them one by one:

1. **Session Cookies Configuration Error**:
   The error message indicates that the `default_redirection_url` is configured globally, which is not supported. Instead, it must be configured per cookie. This means you need to specify the `default_redirection_url` within each cookie configuration block rather than at the global session level. Refer to the session configuration documentation for more details on how to properly configure session cookies: [Session Configuration](https://www.authelia.com/configuration/session/introduction).

2. **OIDC Client Configuration Error**:
   The error messages related to `sector_identifier_uri` indicate that for the OIDC clients 'miniflux' and 'outline-app', the `sector_identifier_uri` is either not provided or is invalid. The `sector_identifier_uri` must be a valid URL with only the host component. If these clients do not require a `sector_identifier_uri`, ensure that this option is either correctly set or omitted if not necessary. For more information on configuring OIDC clients, refer to the OIDC integration documentation: [OIDC Integration](https://www.authelia.com/integration/openid-connect/gitea).

To resolve these issues, you will need to adjust your Authelia configuration file accordingly. For the session cookies, ensure that the `default_redirection_url` is specified within each cookie's configuration block. For the OIDC clients, verify that the `sector_identifier_uri` is correctly configured or omitted if not required.

If these configurations are not present in your configuration file, it's possible that there are defaults or examples in the documentation or sample configurations that are being misunderstood as required. Review your configuration file to ensure it aligns with the requirements outlined in the Authelia documentation.

For further assistance, you can also refer to the Authelia GitHub issues for similar problems or reach out to the Authelia community for support.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
You will need to:

- remove the `default_redirection_url` option, and either add it to the session cookies section or not use it
- remove the blank `sector_identifier_uri` values

**Reply by Dima-Kal** [reply]
As i've mentioned, i do not have sector_identifier_uri anywhere in my config

**Reply by james-d-elliott** [reply, marked as answer]
Do you have `sector_identifier`?

**Reply by Dima-Kal** [reply]
yep that was the one, error should probably mention the correct field name that causing the issue

**Reply by james-d-elliott** [reply]
The complete logs which you did not include tell you that it was remapped.

**Reply by Dima-Kal** [reply]
i only look at errors :)

**Reply by james-d-elliott** [reply]
Something like this:

`"configuration key 'identity_providers.oidc.clients[].sector_identifier' is deprecated in 4.38.0 and has been replaced by 'identity_providers.oidc.clients[].sector_identifier_uri': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"`

**Reply by james-d-elliott** [reply]
That's fine, but please include the full logs and configuration in the future.
